CMakeLists.txt

```diff
@@ -618,6 +618,13 @@ endif()
 # =============================================================================
 set(CODING_CONV_PREFIX "NRN")
 set(${CODING_CONV_PREFIX}_3RDPARTY_DIR "external")
+set(${CODING_CONV_PREFIX}_ClangFormat_REQUIRED_VERSION 12.0.1)
+set(${CODING_CONV_PREFIX}_ClangFormat_EXCLUDES_RE
+    # Exclude translated MOD files that are put in the source tree. This can be
+    # dropped once https://github.com/neuronsimulator/nrn/pull/1673 is merged.
+    ".*?/src/nrnoc/\\(apcount\\|exp2syn\\|expsyn\\|feature\\|hh\\|intfire1\\|intfire2\\|intfire4\\|netstim\\|oclmp\\|passive\\|pattern\\|ppmark\\|stim\\|svclmp\\|syn\\|vclmp\\)\\.c$"
+    ".*?/src/nrnoc/nrnversion\\.h$"
+    CACHE STRING "")
```

Comment:
Should we still rely on list of files generated by CMake? Yesterday I was mentioning [explicit list here](https://github.com/neuronsimulator/nrn/blob/master/cmake/NeuronFileLists.cmake).

If we need to cover all src files without worrying about different build options/targets, should we use above list? (and would that help to cleanup some of extra target dependencies added below?)

Reply:
I think that we should improve the implementation of the formatting check, so that it covers all files in the repository (apart from an explicit list of exceptions), but I propose that this is done later/separately.

When we do that, we could choose to use the explicit lists that you link to, but I am not sure if this is the best approach. If we do that then you still need to run the whole CMake configuration before you can run the formatting check, which feels excessive.

Reply:
I have a feeling this issue will recur several times. Ie which .h files can be formatted.  And I'm unclear about what happens to files like
```
if(MINGW)
  list(APPEND IVOC_FILE_LIST ivocwin.cpp)
else()
```
I guess my question is: Does clang-format do its work correctly regardless of whether a file would normally have to be cross-compiled for successful compiling (I don't see how that can be when there are missing ```#include```) I'm wondering if clang-format is in fact restricted to files that pass the preprocessor stage. I guess I should check to see if formatting reaches into
```
# if 0
...
# endif
```

Reply:
> but I propose that this is done later/separately.

sounds good to me. we could these aspects separately.